modules/ROOT/pages/6.4-release-notes.adoc

```diff
@@ -0,0 +1,1252 @@
+
+//https://ephocks.atlassian.net/browse/TINY-9442
+After {productname} introduced the Smart Typography plugin, a bug was discovered that was causing a space to be introduced between a comma and a closing double-quote mark, as shown in the example below:
+
+.Example
+> “They went that way,[space introduced here]” he said as he pointed.
+
+As demonstrated in the example above, an unintentional or redundant space was inserted into a separate node position without intention.
+
+{productname} fixed the issue by changing the sequence of executing safe and unsafe rules surrounding Typography.
+
+In {productname} version 6.4, the Smart Typography plugin no longer creates a space between a comma and a closing double-quote mark.
+
+=== Added common/space/delBetweenExclamationMarks to remove spaces between exclamation marks
+
+//https://ephocks.atlassian.net/browse/TINY-9398		
+
+An issue was discovered that affected the typography plugin in previous versions of {productname} when inserting content into the editor. Specifically, the issue involved exclamation marks and the spaces between them, such as in the example `“Hello! !“`. This issue prevented the editor from removing the spaces between the double *exclamation marks* `! !` after executing the typography plugin toolbar or menubar button.
+
+The fix for this involved adding a new rule which removes the spaces between exclamation marks when the typography button is pressed.
+
+In {productname} 6.4, spaces are now removed between exclamation marks `!!` for formatted HTML documents when the typography plugin is used. To learn more about Advanced Typography and how to add this new premium plugin to your project, please refer xref:advanced-typography.adoc[Advanced Typography].
+
+=== Added typography plugin to the default menubar
+
+//https://ephocks.atlassian.net/browse/TINY-9450		
+
+In previous versions of {productname}, the typography button would *not* appear in the `menubar` by default. As a consequence, the user could only `find` the typography button in the `toolbar` due to it not appearing in the `menubar` dropdown.
+
+To correct this bug, the typography plugin was added to the default `menubar` config.
+
+{productname} 6.4, by default, will now display the typography button in both the `toolbar` and below the “Source code" button in the `menubar`.
+
+=== Typography button and menu item will be disabled if editor has no content
+
+// https://ephocks.atlassian.net/browse/TINY-9441
+In previous versions of {productname}, the typography plugin button would render to the toolbar when the editor contained no content.
+
+As a consequence of this, users would be able to apply the typography button features to an editor that contained no physical content.
+
+As a fix, {productname} added some new improvements to the Advanced Typography plugin that now disables the users ability to apply the plugin and use its features, when the editor content value returns as `empty`.
```

Comment:
@tiny-ben-tran are you referring to the `execcommands`? and that they could override this disable feature?

Reply:
@tiny-ben-tran this has been tested, and while the `tinymce.activeEditor.execCommand('mceTypography')` can be used to trigger the button, as the editor has no content, this would have no value if the command is executed.